### PR TITLE
nucleotide-count: now both count() and nucleotide_counts() reject invalid input.

### PR DESCRIPTION
**Pull request by orozcoadrian**
added validation to nucleotide_counts(). now both count() and nucleotide_counts() reject invalid input. added a new test and corresponding implementation in the example.py

### CODE REVIEW

**Comment by kytrinyx** [comment]
Thank you, @orozcoadrian. I'll let @Dog and/or @sjakobi review the changes.

Two tiny things

1. Would you mind squashing the commits into a single commit? http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request
2. Take a look at this short blog post about commit messages: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

**Comment by orozcoadrian** [comment]
squashed into a single commit and re-formatted the commit message. thanks.

File: nucleotide-count/example.py [hunk 1]
```diff
@@ -7,6 +7,7 @@ def count(strand, abbreviation):
 
 
 def nucleotide_counts(strand):
+    [_validate(x) for x in strand]
```
Comment (behrtam) [inline nucleotide-count/example.py]: List comprehensions are meant to be used with functions having return values. Just use a `for` loop.

**Comment by orozcoadrian** [comment]
updated the last commit per the review comment. the ci build failure seems unrelated to my changes. let me know if that's something i should look into...

**Comment by behrtam** [comment]
It's not really related to this pr, but why is there a `'U'` in the `NUCLEOTIDES` constant? Doesn't that make the validation incorrect? Based on the [README](https://github.com/exercism/x-common/blob/master/nucleotide-count.md) only 'A', 'C', 'G', and 'T' are nucleotides.

**Comment by orozcoadrian** [comment]
yea i saw that. according to wikipedia, it's "uracil" and it's only found in RNA, not in DNA. i thought about refactoring a bit to minimize the duplication between 'ATGC' and the NUCLEOTIDES variable but decided to go simple for my first pr. i can remove the 'U' and reuse the variable on line 14 if you want.

**Comment by kytrinyx** [comment]
The build is failing on Python 3.4, but I haven't looked into the details of why.

**Comment by Dog** [comment]
I wouldn't validate the strand inside of this function. While its worth considering validation, I don't think it should be necessary for this problem. In my work, I'd usually be validating the strand before calling functions like this.

**Comment by orozcoadrian** [comment]
ok. i can agree with not incorporating this change. i'll keep looking for any contributions i can help with.

**Comment by behrtam** [comment]
You could start with refactoring the `example.py` by removing the `'U'` and reusing the constant on line 14.

**Pull request closed without merging**